x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/ml/datafeed/DatafeedConfig.java

```diff
@@ -676,6 +701,18 @@ public void setDelayedDataCheckConfig(DelayedDataCheckConfig delayedDataCheckCon
             this.delayedDataCheckConfig = delayedDataCheckConfig;
         }
 
+        public void setMaxEmptySearches(int maxEmptySearches) {
+            if (maxEmptySearches == -1) {
```

Comment:
> Since the field is nullable we can pass "Integer" to this method.

This would be true for initial creation of the config, but there is a complication with updates, which is also related to your other question.

The update needs a way to distinguish "I don't want to update this setting" from "I want to update this setting to 'never'".  This needs to be a way that will work via a JSON REST request, not just a Java API.

> Alternatively, we could pass "Optional" or introduce another method "clearMaxEmptySearches".

If the interface was purely Java then these would be better.  But since the REST request for updates will have to have something like `-1` to mean "I want to update this setting to 'never'" I thought it was better to duplicate that here rather than (for example) translate `-1` to an empty optional in the update REST action, then take an optional here.  Doing that would force future maintainers to look through more source files to see how the REST level documentation mapped to the Java code.